packages/contracts/contracts/engine/V3/EngineFactoryV0.sol

```diff
@@ -0,0 +1,215 @@
+    /**
+     * The implementation contract that is cloned when creating new Engine
+     * Flex contracts.
+     */
+    address public immutable engineFlexImplementation;
+
+    // Address of core registry contract.
+    address public immutable coreRegistry;
+
+    /**
+     * Indicates whether the contract is abandoned.
+     * Once abandoned, the contract can no longer be used to create new Engine
+     * and Engine Flex contracts.
+     */
+    bool public isAbandoned; // default false
+
+    /**
+     * @notice validates and assigns immutable configuration variables
+     * @param engineImplementation_ address of the Engine
+     * implementation contract
+     * @param engineFlexImplementation_ address of the Engine Flex
+     * implementation contract
+     * @param coreRegistry_ address of the core registry contract
+     */
+    constructor(
+        address engineImplementation_,
+        address engineFlexImplementation_,
+        address coreRegistry_
+    ) {
+        coreRegistry = coreRegistry_;
+        engineFlexImplementation = engineFlexImplementation_;
+        engineImplementation = engineImplementation_;
+        // set owner of core registry contract
+        Ownable(coreRegistry_).transferOwnership(address(this));
+        // emit event
+        emit Deployed({
+            engineImplementation: engineImplementation_,
+            engineFlexImplementation: engineFlexImplementation_,
+            type_: type_
+        });
+    }
+
+    /**
+     * @notice Creates a new Engine or Engine Flex contract with the provided
+     * `engineConfiguration`, depending on the `engineCoreType`.
+     * @param engineCoreType Type of Engine Core contract.
+     * @param engineConfiguration EngineConfiguration data to configure the
+     * contract with.
+     * @param adminACLContract Address of admin access control contract, to be
+     * set as contract owner. A new contract will be deployed if address is null.
+     * @return engineContract The address of the newly created Engine or Engine Flex
+     * contract. The address is also emitted in both the `EngineCreated` and
+     * `EngineFlexCreated` events.
+     */
+    function createEngineContract(
+        IEngineFactoryV0.EngineCoreType engineCoreType,
+        EngineConfiguration calldata engineConfiguration,
+        address adminACLContract
+    ) external onlyOwner returns (address engineContract) {
+        require(!isAbandoned, "factory is abandoned");
+        // validate engine contract configuration
+        _onlyNonZeroAddress(engineConfiguration.renderProviderAddress);
+        _onlyNonZeroAddress(engineConfiguration.randomizerContract);
+        if (adminACLContract == address(0)) {
+            // deploy new AdminACLV0 contract
+            adminACLContract = Create2.deploy({
+                amount: 0,
+                salt: keccak256(abi.encodePacked(msg.sender, block.timestamp)),
+                bytecode: type(AdminACLV0).creationCode
+            });
```

Comment:
I think you can update superAdmin on the adminACL here instead of later, and just pass an empty array into the core contracts to be updated (since that won't be indexed until later)

Reply:
Updated! Does that get indexed when we call `_transferOwnership(_adminACLContract);`?

Reply:
I believe we only ping the admin ACL for superAdmin in the subgraph when a contract is assigned to it - so as long as we update superAdmin before the contract starts using it, it should index appropriately 😃